src/store/users/saga.test.ts

```diff
@@ -28,8 +28,20 @@ describe(clearUsers, () => {
 
 describe(receiveSearchResults, () => {
   it('translates the search results into user records', async () => {
-    const user1 = { id: 'user-1', name: 'Test User 1', profileImage: 'image-url-1', primaryZID: 'zid-1' };
-    const user2 = { id: 'user-2', name: 'Test User 2', profileImage: 'image-url-2', primaryZID: 'zid-2' };
+    const user1 = {
+      id: 'user-1',
+      name: 'Test User 1',
+      profileImage: 'image-url-1',
+      primaryZID: 'zid-1',
+      wallets: [{ publicAddress: 'address-1' }],
+    };
+    const user2 = {
+      id: 'user-2',
+      name: 'Test User 2',
+      profileImage: 'image-url-2',
+      primaryZID: 'zid-2',
+      wallets: [{ publicAddress: 'address-2' }],
```

Comment:
if a user has a primary_zid set, with a wallet address. then we'll still display the zid in the `subLabel` right (in the search results in the UI)?

Reply:
Yeah this is just setting the users record (which I've now changed slightly to set `displaySubHandle` rather than the wallets, to align with the work you had previously done.

```
export function* receiveSearchResults(searchResults) {
  const existingUsers = (yield select((state) => state.normalized.users)) ?? {};
  const existingUserIds = Object.keys(existingUsers);

  const mappedUsers = searchResults
    .filter((r) => !existingUserIds.includes(r.id))
    .map((r) => {
      return {
        userId: r.id,
        firstName: r.name,
        profileImage: r.profileImage,
        matrixId: r.matrixId,
        primaryZID: r.primaryZID,
        displaySubHandle: getUserHandle(r.primaryZID, r.wallets?.[0]),
      };
    });
  yield put(receive(mappedUsers));
}
```

As for what will be displayed, as we use `getUserHandle` this will determine if its displays the zid or address.

<img width="276" alt="Screenshot 2024-02-08 at 12 29 28" src="https://github.com/zer0-os/zOS/assets/39112648/b17e77de-1c1e-497c-80b3-ef2af87b4cf2">

Reply:
got it. Looks great!